modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -456,7 +456,8 @@ private CompletableFuture<Void> createZonePartitionReplicationNode(
             try {
                 CatalogZoneDescriptor zoneDescriptor = catalogMgr.zone(replicaGrpId.zoneId(), catalogMgr.latestCatalogVersion());
 
-                AtomicLong stamp = new AtomicLong();
+                // if [null], lock wasn't acquired because of exceptions.
+                Long[] stamp = new Long[]{ null };
```

Comment:
Why an array?

Reply:
Because we need to change it from closure and the AtomicLong can't have non-long values, but I need to have a signal, that we didn't set any value. So, it's usual hack - array or AtomicReference, array just a little bit lightweight.